src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2234,5 +2234,24 @@ public override DeconstructionInfo GetDeconstructionInfo(ForEachVariableStatemen
             MemberSemanticModel memberModel = GetMemberModel(node);
             return memberModel?.GetDeconstructionInfo(node) ?? default;
         }
+
+        internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
+        {
+            var location = symbol switch
+            {
+                LocalSymbol { Locations: { IsDefaultOrEmpty: false } locations } => locations[0],
```

Comment:
>{ Locations: { IsDefaultOrEmpty: false } locations } [](start = 28, length = 52)

This code is completed three times. This check could be performed once. It doesn't have to be bundles with symbol kind check.